### PR TITLE
Prevent morphing the home page, when municipality is selected

### PR DESCRIPTION
**Pull request by nmauersberg**
This pr add checks, if we are on a municipality page, when content of the main page / municipality page should change due to the selected municipality in the context.

It is more of a temporary fix, but a more elaborate solution would require more work.

### CODE REVIEW

**Comment by Valioesi** [comment]
I'm still not sure I fully understand this approach. If the municipality is set in the context it means (or rather it should mean) that we are on the municipality page (meaning the check of the url would be redundant). So setting the municipality in the SearchPlaces component would be a problem (which apparently we forgot to remove).

File: src/components/Municipality/MapAndSearch/index.js [hunk 1]
```diff
@@ -23,7 +24,7 @@ export const MapAndSearch = () => {
   }, [leaderboardSegments]);
 
   useEffect(() => {
-    if (municipality) {
+    if (municipality && isMunicipalityPage) {
```
Comment (Valioesi) [inline src/components/Municipality/MapAndSearch/index.js:27]: This leads to the searched municipality not being highlighted.

**Review by Valioesi — CHANGES_REQUESTED** [review]

**Comment by nmauersberg** [comment]
@Valioesi Ah okay, is this the case, that setting the municipality in context means, that we are on the municipality page? I thought we might end up in trouble there, because we have to make sure, it will be reset properly when leaving the page.

File: src/components/Municipality/MapAndSearch/index.js (hunk 1 above)
Comment (nmauersberg) [reply src/components/Municipality/MapAndSearch/index.js]: true! thx. It's fixed

**Comment by Valioesi** [comment]
Good point with the resetting. Isn't it done anywhere? In that case the whole municipality in context approach was not ideal to begin with.

**Comment by nmauersberg** [comment]
Yeah I think it was a valid idea at some point, but it did not scale very well. It is so hard to debug. Maybe this PR could be a temporary fix and we replace the whole approach with a better one?

At the moment I still have the issue, that when I am logged in, the Leaderboard vanishes as soon as I select a municipality. I've been looking for 30 min. to find the culprit already :D

**Comment by nmauersberg** [comment]
Resetting was done in the context, on navigating pages. I think it was removed by the first effort to stop the page from morphing back and forth. I'll look into that again.

**Comment by Valioesi** [comment]
I am fine with it being a temporary fix. I just thought it might be confusing in the future, when people don't know when they need to use which check. But yeah, maybe we can come up with something better entirely. 

I just check on the local build. When I navigate to a municipality page and then to /modellversuch municipality is undefined again.

**Comment by nmauersberg** [comment]
@Valioesi Ah cool, that might be better! thx for checking!

**Review by Valioesi — APPROVED** [review]

**Pull request closed without merging**